pkg/components/cluster-autoscaler/component.go

```diff
@@ -139,37 +139,85 @@ func newComponent() *component {
 	return c
 }
 
-// getWorkerUserdata finds a worker from clusterName in facility given a list
-// of devices in a project and returns its user data. If two devices with the
-// same name are found it returns an error.
-func getWorkerUserdata(clusterName, facility string, devices []packngo.Device) (string, error) {
-	var userData string
-	deviceSet := make(map[string]struct{})
+// getClusterWorkers takes a list of devices from the user and returns list
+// of worker nodes belonging to the specified cluster name and facility.
+func getClusterWorkers(clusterName, facility string, devices []packngo.Device) []packngo.Device {
+	clusterWorkers := []packngo.Device{}
 
 	for _, d := range devices {
+		// Skip devices from other facilities.
 		if d.Facility.Code != facility {
 			continue
 		}
 
-		if _, ok := deviceSet[d.Hostname]; !ok {
-			deviceSet[d.Hostname] = struct{}{}
-		} else {
-			return "", fmt.Errorf("having two devices with the same name (%q) in the same facility is not supported", d.Hostname)
+		// Skip devices from other clusters.
+		if !strings.Contains(d.Hostname, clusterName) {
+			continue
 		}
 
-		// if device hostname contains the cluster name and "worker", we want
-		// its user data
-		if strings.Contains(d.Hostname, clusterName) &&
-			strings.Contains(d.Hostname, "worker") {
-			userData = base64.StdEncoding.EncodeToString([]byte(d.UserData))
+		// Skip controller nodes.
+		if !strings.Contains(d.Hostname, "worker") {
```

Comment:
Just curious why the check is `not wokers` vs `controllers`?

Reply:
Well, there could possibly be other type of nodes in the future :smile: also this was in the original code. Maybe I change the commit to `Skip non-worker nodes.` ?

Reply:
Changed the comment.